File: ts/src/metadata/tile/logicalStreamType.ts
import type { DictionaryType } from "./dictionaryType";
import type { OffsetType } from "./offsetType";
import type { LengthType } from "./lengthType";

export type LogicalStreamType = {
    readonly dictionaryType?: DictionaryType;
    readonly offsetType?: OffsetType;
    readonly lengthType?: LengthType;
}

File: java/src/main/java/org/maplibre/mlt/metadata/stream/LogicalStreamType.java
package org.maplibre.mlt.metadata.stream;

public class LogicalStreamType {
  private DictionaryType dictionaryType;
  private OffsetType offsetType;
  private LengthType lengthType;

  public LogicalStreamType(DictionaryType dictionaryType) {
    this.dictionaryType = dictionaryType;
  }

  public LogicalStreamType(OffsetType offsetType) {
    this.offsetType = offsetType;
  }

  public LogicalStreamType(LengthType lengthType) {
    this.lengthType = lengthType;
  }

  public DictionaryType dictionaryType() {
    return this.dictionaryType;
  }

  public OffsetType offsetType() {
    return this.offsetType;
  }

  public LengthType lengthType() {
    return this.lengthType;
  }
}
